### PR TITLE
[docs] update readme with bypass device instructions

### PR DESCRIPTION
**Pull request by jarrodparkes**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Quick backstory — I manage Apple developer accounts for ~100 different customers, some organization accounts, some individual accounts. For the individual accounts, I'm forced to access the Apple developer portal using their credentials which means 2FA in most cases. Thankfully, by coordinating with the customer, I can add my phone as a trusted device so that I don't have to contact them each time I need to access their account or generate a session...

But, the problem arises when I try to generate a web session using `fastlane spaceauth -u [apple_id_email]`. Since most of these Apple accounts have devices other than my phone registered, fastlane never prompts me to choose my phone for sending the code. Looking at the documentation and this closed issue #10790, there is no way to force fastlane to send the code to a certain number. One option would be to remove all the registered devices from the account, leaving only the desired phone number, but this is problematic because many of the customers I work with are using their devices with Apple throughout the day and logging them out would be BAD. The other option is to take the route recommended in this updated documentation:

- Attempt to generate a web session with `fastlane spaceauth -u apple@krausefx.com` and wait for security code prompt to appear
- Open a browser to appleid.apple.com or an address that requires you to login with your Apple ID; logout of any previous session
- Login with your Apple ID, and when prompted for a security code, request a code be sent to the desired phone
- Use the code sent to phone with fastlane instead of with the browser

It isn't an open issue, but it still helps with this closed issue https://github.com/fastlane/fastlane/issues/10790.

### Description

Changed the documentation to include a way to bypass trusted devices and use a trusted phone number instead.

See the steps listed in the documentation. I've been able to reproduce the steps, and I've successfully generate a web session multiple times using this method.

### CODE REVIEW

**Comment by jarrodparkes** [comment]
I signed it!

**Comment by jarrodparkes** [comment]
let me know if you have any questions on the PR 😄

**Comment by KrauseFx** [comment]
This looks great on the first look, thanks for your contribution. This should also be applied over here, right? https://github.com/fastlane/docs/blob/master/docs/best-practices/continuous-integration.md

**Comment by jarrodparkes** [comment]
cool, didn't know that existed. yep, let me add something there too

**Comment by jarrodparkes** [comment]
@KrauseFx added similar doc changes in https://github.com/fastlane/docs/pull/729

File: spaceship/README.md [hunk 1]
```diff
@@ -133,6 +133,15 @@ export FASTLANE_SESSION='---\n- !ruby/object:HTTP::Cookie\n  name: DES5c148586df
 
 Copy everything from `---\n` to your CI server and provide it as environment variable named `FASTLANE_SESSION`.
 
+#### Bypass device and use SMS for verification
```
Comment (joshdholtz) [inline spaceship/README.md]: Can we change this to...
```md
#### Bypass trusted device and use SMS for verification
```

File: spaceship/README.md [hunk 2]
```diff
@@ -133,6 +133,15 @@ export FASTLANE_SESSION='---\n- !ruby/object:HTTP::Cookie\n  name: DES5c148586df
...
+
+If you have a trusted device configured for your Apple account, then Apple will not send a SMS code to your phone when you try to generate a web session with fastlane — instead a code will be displayed on one of your account's trusted devices. This can be problematic if you are trying to authenticate, but don't have access to a device. To circumvent the device and use SMS instead, take the following steps:
```
Comment (joshdholtz) [inline spaceship/README.md]: Can we reword this to the following?
```md
Apple will not send a SMS code to your phone if you have a trusted device configured for your Apple account when you try to generate a web session with fastlane — instead a code will be displayed on one of your account's trusted devices. This can be problematic if you are trying to authenticate but don't have access to a trusted device. Take the following steps to circumvent the device and use SMS instead:
```

File: spaceship/README.md [hunk 3]
```diff
@@ -133,6 +133,15 @@ export FASTLANE_SESSION='---\n- !ruby/object:HTTP::Cookie\n  name: DES5c148586df
...
+
+- Attempt to generate a web session with `fastlane spaceauth -u apple@krausefx.com` and wait for security code prompt to appear
```
Comment (joshdholtz) [inline spaceship/README.md]: Can we replace `apple@krausefx.com` with `[email]`? I don't want somebody accidentally trying to auth in as Felix 😛

File: spaceship/README.md [hunk 4]
```diff
@@ -133,6 +133,15 @@ export FASTLANE_SESSION='---\n- !ruby/object:HTTP::Cookie\n  name: DES5c148586df
...
+- Open a browser to appleid.apple.com or an address that requires you to login with your Apple ID, and logout of any previous session
```
Comment (joshdholtz) [inline spaceship/README.md]: Can we change `appleid.apple.com` to `[appleid.apple.com](https://appleid.apple.com)` so that we get a nice link?

File: spaceship/README.md [hunk 5]
```diff
@@ -133,6 +133,15 @@ export FASTLANE_SESSION='---\n- !ruby/object:HTTP::Cookie\n  name: DES5c148586df
...
+- Login with your Apple ID, and when prompted for a security code, request a code be sent to the desired phone
```
Comment (joshdholtz) [inline spaceship/README.md]: Can we reword this to...
```md
- Login with your Apple ID and request a code be sent to the desired phone when prompted for a security code 
```

File: spaceship/README.md [hunk 6]
```diff
@@ -133,6 +133,15 @@ export FASTLANE_SESSION='---\n- !ruby/object:HTTP::Cookie\n  name: DES5c148586df
...
+- Use the code sent to phone with fastlane instead of with the browser
```
Comment (joshdholtz) [inline spaceship/README.md]: Can you add underscores around "fastlane" in here? `_fastlane_`

File: spaceship/README.md (hunk 2 above)
Comment (joshdholtz) [reply spaceship/README.md]: Can you also add underscores around "fastlane" in here too? `_fastlane_`?

**Review by joshdholtz — CHANGES_REQUESTED** [review]
This is a great addition! ❤️ Just left a few comments/requests on some rewording and formatting 🙃

**Comment by jarrodparkes** [comment]
@joshdholtz done! thanks for taking a look :)

File: spaceship/README.md [hunk 7]
```diff
@@ -133,6 +133,15 @@ export FASTLANE_SESSION='---\n- !ruby/object:HTTP::Cookie\n  name: DES5c148586df
 
 Copy everything from `---\n` to your CI server and provide it as environment variable named `FASTLANE_SESSION`.
 
+#### Bypass trusted device and use SMS for verification
+
+Apple will not send a SMS code to your phone if you have a trusted device configured for your Apple account when you try to generate a web session with _fastlane_ — instead a code will be displayed on one of your account's trusted devices. This can be problematic if you are trying to authenticate but don't have access to a trusted device. Take the following steps to circumvent the device and use SMS instead:
```
Comment (joshdholtz) [inline spaceship/README.md]: I read through this one more time and think ONE more rework will make this 💯 😊

>If you have a trusted device configured, Apple will not send a SMS code to your phone for your Apple account when you try to generate a web session with _fastlane_. Instead, a code will be displayed on one of your account's trusted devices. This can be problematic if you are trying to authenticate but don't have access to a trusted device. Take the following steps to circumvent the device and use SMS instead:

**Review by joshdholtz — CHANGES_REQUESTED** [review]
One more small rework 😊 Can you also make these same changes over on https://github.com/fastlane/docs/blob/master/docs/best-practices/continuous-integration.md as well? 🙃

This should be good to merge after this!

**Comment by jarrodparkes** [comment]
updated the paragraph here, and made identical changes on https://github.com/fastlane/docs/pull/729

**Review by joshdholtz — APPROVED** [review]
Looks good! Thank you 😊

**Comment by janpio** [comment]
Is there a specific reason why you log out from Apple only _after_ you got the "push" 6-digit code, or could steps 1 and 2 just be switched?

**Comment by jarrodparkes** [comment]
@janpio not that i can think of. i believe you could switch those steps without issue. have you tried doing that yourself?

**Comment by janpio** [comment]
Yeah, I am currently looking into https://github.com/fastlane/fastlane/pull/13554 and https://github.com/fastlane/fastlane/pull/13507 (which automate this process and integrate this into the normal flow) and was just wondering if this might have any influence.

**Comment by jarrodparkes** [comment]
@janpio gotcha. i think my initial thinking on having the prompt display first (before attempting to login via the browser) was that perhaps the layers underlying the login mechanism would detect and only accept codes from where the initial request came from. however, if your testing suggests otherwise, then it probably doesn't matter. i think i've swapped steps 1 and 2 before without issue, but id be curious to know what you see 😄
